app/models/user.rb

```diff
@@ -63,4 +65,8 @@ def self.new_with_session(params, session)
       end
     end
   end
+
+  def watching?(task)
+    self.tasks.include?(task)
```

Comment:
What SQL query it generates?
change to active record. Does it return all tasks as relation?

use "where(...).any?"